### ISSUE TITLE
Previous not populated when using --tag and --unreleased

### ISSUE THREAD

**Issue by iCrawl** [orhun/git-cliff#47]
**Describe the bug**
`git cliff --unreleased --tag 14.0.0 --prepend ./CHANGELOG.md`

If a repo is needed:
https://github.com/discordjs/discord.js

Running:
`cd packages/discord.js`
`git cliff --unreleased --tag 14.0.0 --prepend ./CHANGELOG.md -c ../../cliff.toml -r ../../`

Will reproduce it, otherwise similar steps will most likely work too on any repo.

**To Reproduce**
```toml
[changelog]
header = """
# Changelog
All notable changes to this project will be documented in this file.\n
"""
body = """
{% if version %}\
    # [{{ version | trim_start_matches(pat="v") }}]\
    {% if previous %}\
        {% if previous.version %}\
            (https://github.com/discordjs/discord.js/compare/{{ previous.version }}...{{ version }})\
        {% else %}
            (https://github.com/discordjs/discord.js/tree/{{ version }})\
        {% endif %}\
    {% endif %} \
    - ({{ timestamp | date(format="%Y-%m-%d") }})
{% else %}\
    # [unreleased]
{% endif %}\
{% for group, commits in commits | group_by(attribute="group") %}
    ## {{ group | upper_first }}
    {% for commit in commits %}
        - {% if commit.breaking %}\
            [**breaking**] \
          {% endif %}\
            {% if commit.scope %}\
                **{{commit.scope}}:** \
            {% endif %}\
            {{ commit.message | upper_first }} ([{{ commit.id | truncate(length=7, end="") }}](https://github.com/discordjs/discord.js/commit/{{ commit.id }}))\
    {% endfor %}
{% endfor %}\n
"""
trim = true
footer = ""

[git]
conventional_commits = true
filter_unconventional = true
commit_parsers = [
    { message = "^feat", group = "Features"},
    { message = "^fix", group = "Bug Fixes"},
    { message = "^docs", group = "Documentation"},
    { message = "^perf", group = "Performance"},
    { message = "^refactor", group = "Refactor"},
    { message = "^typings", group = "Typings"},
    { message = "^types", group = "Typings"},
    { message = ".*deprecated", body = ".*deprecated", group = "Deprecation"},
    { message = "^revert", skip = true},
    { message = "^style", group = "Styling"},
    { message = "^test", group = "Testing"},
    { message = "^chore", skip = true},
    { message = "^ci", skip = true},
    { message = "^build", skip = true},
    { body = ".*security", group = "Security"},
]
filter_commits = true
tag_pattern = "[0-9]*"
skip_tags = "v[0-9]*|11|12"
ignore_tags = ""
topo_order = false
sort_commits = "newest"
```

**Expected behavior**
Expect `previous` to be populated to based on the latest actual tag so the desired compare link can be generated with the config above.

**Screenshots/Logs**
Generated by creating a git tag manually and then running: `git cliff --latest --prepend ./CHANGELOG.md`
![image](https://user-images.githubusercontent.com/20760160/150020962-f48ae96d-4ba1-41e5-b7fc-c59636ff5980.png)

Generated by not creating a git tag and running `git cliff --unreleased --tag 14.0.0 --prepend ./CHANGELOG.md`
![image](https://user-images.githubusercontent.com/20760160/150021026-47915e84-d520-43cd-a0d5-ce942e038cf6.png)

**System (please complete the following information):**
 - OS Information: [Windows 11 Pro / 10.0.22000 Build 22000]
 - Project Version: [dev-build 8bb48b09ef4488e4fb6b03a43e1d862b4645971b]

**Comment by orhun** [comment]
This should be fixed in 2be04f8, can you try it out and let me know if it works as expected now?

**Comment by iCrawl** [comment]
Just tested it, works perfectly.

Thank you!
